adapter/src/adapter/api/rest/adapterEndpoint.ts

```diff
@@ -45,13 +49,43 @@ export class AdapterEndpoint {
     res: express.Response,
   ): Promise<void> => {
     const validator = new AdapterConfigValidator();
-    if (!validator.validate(req.body)) {
+    const adapterconfigforValidator = req.body;
+    if (!validator.validate(adapterconfigforValidator)) {
       res.status(400).json({ errors: validator.getErrors() });
       return;
     }
+    // Check protocol type
+    let protocolType = AdapterEndpoint.getProtocol(req.body.protocol.type)
```

Comment:
Pls use `const` instead of `let` for variable declaration where possible.
https://www.tutorialsteacher.com/typescript/typescript-variable